File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        ## edge cases
        if(len(nums) == 0):
            return False        

        ## create a set
        s = set()

        ## check if a number exists in set
        for i in nums:
            if i in s:
                return True
            else:
                s.add(i)

        ## return output
        return False

        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        // edge cases

        //declarations
        int[] result = new int[nums.Length];
        
        // prefix
        int prefix = 1;
        
        for(int i=0; i<nums.Length; i++) {
            result[i] = prefix;
            prefix = prefix * nums[i];
        }
        // postfix
        int postfix = 1;

        for(int i=nums.Length-1; i>=0; i--) {
            result[i] = result[i] * postfix;
            postfix = postfix * nums[i];
        }

        return result;
    }
}
